### PR TITLE
Run java test specs in parallel

### PR DESCRIPTION
**Pull request by amitkrout**
## Was the change discussed in an issue?
fixes - part of #1473

## How to test changes?

make test-java-e2e

### CODE REVIEW

**Comment by amitkrout** [comment]
Blocked due to #1846

**Comment by amitkrout** [comment]
'''
odo generic when .odoignore file exists 
  should create and push the contents of a named component excluding the contents in .odoignore file
  /go/src/github.com/openshift/odo/tests/integration/generic_test.go:104
 Creating a new project: iyirwywxlc
Running odo with args: [project create iyirwywxlc -w -v4] 
[odo] I0705 05:27:00.625993   12354 preference.go:116] The configFile is /tmp/artifacts/.odo/preference.yaml
[odo] I0705 05:27:00.626054   12354 occlient.go:455] Trying to connect to server api.ci-op-cxyqp9s0-f09f4.origin-ci-int-aws.dev.rhcloud.com:6443
[odo] I0705 05:27:00.640618   12354 occlient.go:462] Server https://api.ci-op-cxyqp9s0-f09f4.origin-ci-int-aws.dev.rhcloud.com:6443 is up
[odo] I0705 05:27:00.723231   12354 occlient.go:385] isLoggedIn err:  <nil> 
[odo]  output: "developer"
[odo] I0705 05:27:00.723263   12354 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0705 05:27:00.723271   12354 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: context
[odo] I0705 05:27:00.723309   12354 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0705 05:27:00.723317   12354 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: project
[odo] I0705 05:27:00.744462   12354 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: app
[odo] I0705 05:27:00.744491   12354 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: output
[odo] I0705 05:27:00.744498   12354 context.go:355] Ignoring error as it usually means flag wasn't set: flag accessed but not defined: component
[odo]  •  Waiting for project to come up  ...
[odo]  ✓  Waiting for project to come up [425ms]
[odo]  ✓  Project 'iyirwywxlc' is ready for use
[odo]  ✓  New project created and now using project : iyirwywxlc
[odo] I0705 05:27:01.178295   12354 odo.go:70] Could not get the latest release information in time. Never mind, exiting gracefully :)
Created dir: /tmp/366751980
Running git with args: [clone https://github.com/openshift/nodejs-ex /tmp/366751980/nodejs-ex] 
[git] Cloning into '/tmp/366751980/nodejs-ex'...
Running odo with args: [create nodejs nodejs --project iyirwywxlc --context /tmp/366751980/nodejs-ex] 
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [319ms]
[odo] Please use `odo push` command to create the component with source deployed
Running odo with args: [push --context /tmp/366751980/nodejs-ex] 
[odo] Validation
[odo]  •  Validating component  ...
[odo]  ✓  Validating component [403ms]
[odo]  •  Checking component  ...
[odo]  ✓  Checking component [18ms]
[odo] 
[odo] Configuration changes
[odo]  •  Creating component  ...
[odo]  ✓  Initializing component
[odo]  ✓  Creating component [183ms]
[odo]  •  Applying configuration  ...
[odo]  ✓  Applying configuration [10365ns]
[odo] 
[odo] Pushing to component nodejs of type local
[odo]  •  Waiting for component to start  ...
[odo]  ✓  Waiting for component to start [3m]
[odo]  •  Copying files to component  ...
[odo]  ✓  Copying files to component [34s]
[odo]  •  Building component  ...
[odo]  ✓  Building component [13s]
[odo]  ✓  Changes successfully pushed to component
Running oc with args: [get pods --namespace iyirwywxlc] 
[oc] NAME                  READY   STATUS    RESTARTS   AGE
[oc] nodejs-app-1-deploy   1/1     Running   0          3m38s
[oc] nodejs-app-1-fzt56    1/1     Running   0          3m28s
Running oc with args: [exec nodejs-app-1-deploy --namespace iyirwywxlc -- ls -lai /opt/app-root/src] 
[oc] error: unable to upgrade connection: container not found ("deployment")
Deleting project: iyirwywxlc
Running odo with args: [project delete iyirwywxlc -f] 
[odo] This project contains the following applications, which will be deleted
[odo] Application app
[odo] This application has following components that will be deleted
[odo] component named nodejs
[odo] No services / could not get services
[odo]  •  Deleting project iyirwywxlc  ...
[odo]  ✓  Deleting project iyirwywxlc [6s]
[odo] Deleted project : iyirwywxlc
 • Failure [229.293 seconds]
odo generic
/go/src/github.com/openshift/odo/tests/integration/generic_test.go:16
  when .odoignore file exists
  /go/src/github.com/openshift/odo/tests/integration/generic_test.go:92
    should create and push the contents of a named component excluding the contents in .odoignore file [It]
    /go/src/github.com/openshift/odo/tests/integration/generic_test.go:104
     No future change is possible.  Bailing out early after 0.338s.
      
    Expected
        <int>: 1
    to match exit code:
        <int>: 0
     /go/src/github.com/openshift/odo/tests/helper/helper_run.go:29
'''
Seems a new flake is introduced

**Comment by amitkrout** [comment]
[odo]  ✗  Failed to create component with name wo-wait-javaee-git-test. Please use `odo config view` to view settings used to create component. Error: unable to wait for build wo-wait-javaee-git-test-app-1 to run: timed out waiting for the condition


/test integration

File: tests/integration/java_test.go [hunk 1]
```diff
@@ -38,72 +41,75 @@ var _ = Describe("odoJavaE2e", func() {
...
 			// Ping said URL
 			helper.HttpWaitFor(routeURL, "Hello World", 90, 1)
 
 			// Delete the component
-			helper.CmdShouldPass("odo", "delete", "uberjar-git-test", "-f")
+			helper.CmdShouldPass("odo", "delete", "uberjar-git-test", "-f", "--context", context)
 		})
 
-		It("Should be able to deploy a spring boot uberjar file using openjdk", func() {
+		// https://github.com/openshift/odo/issues/1846
```
Comment (mohammedzee1000) [inline tests/integration/java_test.go]: I am assuming these tests do not pass as of now?

**Comment by amitkrout** [comment]
> As we add context flag to enable parallel testing we should ensure that we are doing atleast one test without context flag :)
> Anycase looks good to me
> /approve

Thanks for the review. 
The combination of with and without context flag will be implemented in #1845

File: tests/integration/java_test.go (hunk 1 above)
Comment (amitkrout) [reply tests/integration/java_test.go]: Yes, you are right.

Comment (mohammedzee1000) [reply tests/integration/java_test.go]: It would be a good idea to record this as an issue so we remember to fix it

**Pull request merged**